src/pybind/mgr/telemetry/module.py

```diff
@@ -885,7 +839,10 @@ def compile_report(self, channels: Optional[List[str]] = None) -> Dict[str, Any]
 
         if 'perf' in channels:
             report['perf_counters'] = self.gather_perf_counters()
-            report['stat_sum_per_pool'] = self.get_stat_sum_per_pool()
+
+            report['stats_per_pool'] = self.get('pg_dump')['pool_stats']
+            report['stats_per_pg'] = self.get('pg_dump')['pg_stats']
```

Comment:
I would separate these changes into 2 commits: one with changes related only to `stats_per_pool`, and another with the new `report['stats_per_pg']` metrics collection.